src/encdatavault_fmt_plug.c

```diff
@@ -0,0 +1,411 @@
+	int nb_keys;
+
+	memset(cracked, 0, sizeof(cracked[0]) * cracked_count);
+	if (1 == cur_salt->version) {
+		nb_keys = 1 << (cur_salt->algo_id - 1);
+	} else {
+		nb_keys = 8;
+	}
+
+#ifdef _OPENMP
+	#pragma omp parallel for
+#endif
+	for (index = 0; index < count; index += MIN_KEYS_PER_CRYPT) {
+		int j;
+		MD5_CTX ctx;
+		unsigned char kdf_out[8][ENC_KEY_SIZE];
+		unsigned char ivs[8][AES_BLOCK_SIZE] = { 0 };
+		unsigned char hash[ENC_KEY_SIZE];
+		unsigned char result[ENC_KEY_SIZE * 8] = { 0 };
+		unsigned char *result_ptr;
+
+		// Key derivation
+		MD5_Init(&ctx);
+		MD5_Update(&ctx, saved_key[index], strlen(saved_key[index]));
+		MD5_Final(hash, &ctx);
+
+		for (j = 0; j < nb_keys; j++) {
+			memcpy(kdf_out[j], default_salts[j], ENC_SALT_SIZE);
+		}
+
+		for (j = 1; j < ENC_DEFAULT_NUM_HASH_ITERATIONS; j++) {
+			MD5_Init(&ctx);
+			MD5_Update(&ctx, hash, 16);
+			MD5_Final(hash, &ctx);
+			enc_xor_block((uint64_t *) result, (uint64_t *) hash);
+		}
+
+		for (j = 0; j < nb_keys; j++) {
+			enc_xor_block((uint64_t *) kdf_out[j], (uint64_t *) result);
+		}
+
+		/* AES CTR */
+		if (cur_salt->version == 1) {
+			enc_xor_block((uint64_t *) ivs[0], (uint64_t *) cur_salt->iv);
+			for (j = 1; j < nb_keys; j++) {
+				enc_xor_block((uint64_t *) ivs[j], (uint64_t *) kdf_out[j]);
+				enc_xor_block((uint64_t *) ivs[j], (uint64_t *) cur_salt->iv);
+			}
+			enc_aes_ctr_iterated(cur_salt->encrypted_data, result, kdf_out[0],
+			                     ivs, AES_BLOCK_SIZE, nb_keys, 1);
+			result_ptr = result + 4;
+			cracked[index] =
+			    (0 == memcmp(result_ptr, "\xd2\xc3\xb4\xa1\x00\x00\x00\x01",
+			                 ENC_SIG_SIZE));
+		} else {
+			// Decrypt keychain
+			for (j = 1; j < nb_keys; j++) {
+				enc_xor_block((uint64_t *) ivs[j],
+				              (uint64_t *) kdf_out[nb_keys - j]);
+			}
+			enc_aes_ctr_iterated(cur_salt->keychain, result, kdf_out[0], ivs,
+			                     ENC_KEYCHAIN_SIZE, nb_keys, 0);
+
+			// Decrypt data
+			enc_xor_block((uint64_t *) ivs[0], (uint64_t *) ivs[0]);
+			enc_xor_block((uint64_t *) ivs[0], (uint64_t *) cur_salt->iv);
+			for (j = 1; j < nb_keys; j++) {
+				enc_xor_block((uint64_t *) ivs[j], (uint64_t *) ivs[j]);
+				enc_xor_block((uint64_t *) ivs[j],
+				              (uint64_t *)(result + j * 16));
+				enc_xor_block((uint64_t *) ivs[j], (uint64_t *) cur_salt->iv);
+			}
+			enc_aes_ctr_iterated(cur_salt->encrypted_data, result, result, ivs,
+			                     AES_BLOCK_SIZE, nb_keys, 1);
+			result_ptr = result + 4;
+			cracked[index] = (0 == memcmp(result_ptr, "\xd2\xc3\xb4\xa1", 4));
+		}
+
+	}
+	return count;
+}
+
+static int cmp_all(void *binary, int count)
+{
+	int index;
+
+	for (index = 0; index < count; index++)
+		if (cracked[index])
+			return 1;
+	return 0;
+}
+
+static int cmp_one(void *binary, int index)
+{
+	return cracked[index];
+}
+
+static int cmp_exact(char *source, int index)
+{
+	return cracked[index];
+}
+
+static void encdv_set_key(char *key, int index)
```

Comment:
Don't prefix the function name with format name here. This function isn't any special compared to other format methods, where you already don't use any prefix.